dask/dataframe/core.py

```diff
@@ -336,6 +335,25 @@ def quantiles(self, q):
         """
         return quantiles(self, q)
 
+    def resample(self, rule, how=None, axis=0, fill_method=None, closed=None,
+                 label=None, convention='start', kind=None, loffset=None,
+                 limit=None, base=0):
+        first, last = self.divisions[0], self.divisions[-1]
+        new_divisions = list(pd.date_range(start=first, end=last, freq=rule))
```

Comment:
Something like that feels optimal, yes.  We haven't greatly change the amount of data in any partition.

Although I think that new_divions should actually be

   1st instant of january, 1st instant of february, 1st instant of october, 1st instant of november

which I suspect is the same as 

    '2014-01', '2014-02', ...

We would not want divisions to be

    last minute of December, last minute of January, last minute of September, last minute of October,